src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/CommonFunction.java

```diff
@@ -84,7 +86,8 @@ public static boolean isApplicantSolicitorAgreeToReceiveEmails(Map<String, Objec
     }
 
     public static boolean isRespondentRepresentedByASolicitor(Map<String, Object> caseData) {
-        return isNotEmpty(RESP_SOLICITOR_NAME, caseData);
+        return YES_VALUE.equalsIgnoreCase(nullToEmpty(caseData.get(CONSENTED_RESPONDENT_REPRESENTED)))
+            || YES_VALUE.equalsIgnoreCase(nullToEmpty(caseData.get(CONTESTED_RESPONDENT_REPRESENTED)));
```

Comment:
updated from checking the name as its unreliable

https://gateway-ccd.aat.platform.hmcts.net/print/jurisdictions/DIVORCE/case-types/FinancialRemedyMVP2/cases/1598514433842461

this shows a case with respondentRepresented: no, but a name in the field.

This can be achieved through the "Update contact details. A case originally represented but swapped to not represented will retain the fields for the solicitor such as name and address.